subprojects/packagefiles/theora/meson.build

```diff
@@ -0,0 +1,269 @@
+project(
+  'theora',
+  'c',
+  'cpp',
+  version: '1.1.1',
+  meson_version: '>= 0.59.0',
+  license: 'BSD-3-Clause',
+  default_options: ['c_std=c11'],
+)
+
+conf = configuration_data()
+
+enable_valgrind = get_option('valgrind-testing').allowed()
+
+test_env = []
+
+if enable_valgrind
+  valgrind_exe = find_program('valgrind')
+
+  conf.set('HAVE_VALGRIND', valgrind_exe.found())
+
+  if conf.get('HAVE_VALGRIND')
+    test_env += [
+      '-q',
+      '--leak-check=yes',
+      '--show-reachable=yes',
+      '--num-callers=100',
+    ]
+
+    message(
+      'Tests will be run under: @0@'.format(env.get('VALGRIND_ENVIRONMENT')),
+    )
+  else
+    message('Valgrind not found')
+  endif
+else
+  message('Valgrind testing not enabled')
+endif
+
+compile_args = []
+
+cc = meson.get_compiler('c')
+
+if cc.get_argument_syntax() != 'msvc'
+  compile_args += cc.get_supported_arguments('-Wall', '-Wno-parentheses')
```

Comment:
I don't see where this variable is used, hmm...

(Also, -Wall is handled by `-Dwarning_level=1`, the default warning level.)

Reply:
Hm, these should be passed on to the libraries in lib, must've slipped when I rebased the patches for 1.1.x...